### DISCUSSION TITLE
New iCloud end to end encryption & new iCloud Data not on Web Option - Breaks script?

### DISCUSSION THREAD

**Discussion by hsingh314** [icloud-photos-downloader/icloud_photos_downloader#509] [Q&A] [answered]
Hi, 

So I tried backing up last night and in the middle of the night it stopped and when I resumed it today, I got the "access denied" error. I tried to go to icloud.com and nothing throughout the day, but looking at apple's service status during the day, iCloud services were going up and down. However, just now my phones and computer got pushed a new update which included an an option in iCloud for advanced data protection (end to end encryptions, and only certain devices can log in) and the default is to not allow icloud to be viewed from the web (icloud.com doesn't have any photos, notes, mail). 

I was wondering if maybe this is why I got access denied errors before and if anyone else is having issues with icloud and backing up right now also? 

Am I to assume to turn off advanced data protection and to turn on allow icloud on the web?

**Comment by boredazfcuk** [comment, marked as answer]
If you enable Advanced Data Protection, you cannot use this script. ADP and icloudpd are mutually exclusive.

See here: https://support.apple.com/en-us/HT212520

>Web access to your data at iCloud.com
>When you turn on Advanced Data Protection, access to your iCloud data on the web at iCloud.com is disabled

This script automates logging in to icloud.com and downloading the photos. Enabling Advanced Data Protection means you are disabling the access vector required by the script.

If you wish to disable ADP, instructions on how to do this are at the bottom of the link above.
